ext/mbstring/libmbfl/filters/mbfilter_cjk.c

```diff
@@ -11560,6 +11560,45 @@ static void mb_wchar_to_cp936(uint32_t *in, size_t len, mb_convert_buf *buf, boo
 	MB_CONVERT_BUF_STORE(buf, out, limit);
 }
 
+/* Step through a GB18030 string one character at a time. Find the last position at or
+ * before `limit` which falls directly after the end of a (single or multi-byte) character */
+static zend_always_inline unsigned char* step_through_gb18030_str(unsigned char *p, unsigned char *limit)
+{
+	while (p < limit) {
+		unsigned char c = *p;
+		if (c < 0x81 || c == 0xFF) {
```

Comment:
It looks like 0x80 is an invalid codepoint same as `0xFF`, is thy why we just "ignore" like ASCII?

Reply:
Well, the important point for this line of code is that 0xFF does not initiate a 2-byte or 4-byte character in GB18030.

Reply:
ACK